### ISSUE TITLE
TEM EDS model plot drawing excessively tall image when y-axis is logarithmic and including line labels

### ISSUE THREAD

**Issue by ZGainsforth** [hyperspy/hyperspy#2991]
I have an EDS TEM spectrum which I have assigned lines and backgrounds to and fit.  It generally works well.  With some spectra, peaks are fit at very low values close to zero (because the peaks really aren't there!) and when plotting a logarithmic y-axis I get a crash:

`ValueError: Image size of 1144x97463 pixels is too large. It must be less than 2^16 in each direction.`

It seems to pertain to the rescaling of the y-axis because this fails:

```
# Plot it.
m.isig[0.0:12.0].plot(norm='log', xray_lines=True)
```

And this also fails:

```
# Plot it.
m.plot(norm='log', xray_lines=True)
ax=plt.gca()
ax.set_ylim(0.1,np.max(S.data))
ax.set_xlim(0,12)
```

But this succeeds (but draws a really tall image):

```
# Plot it.
m.plot(norm='log', xray_lines=True)
ax=plt.gca()
ax.set_xlim(0,12)
```

The full error is:

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
File ~\anaconda3\envs\conda310\lib\site-packages\IPython\core\formatters.py:339, in BaseFormatter.__call__(self, obj)
    337     pass
    338 else:
--> 339     return printer(obj)
    340 # Finally look for special method names
    341 method = get_real_method(obj, self.print_method)

File ~\anaconda3\envs\conda310\lib\site-packages\IPython\core\pylabtools.py:151, in print_figure(fig, fmt, bbox_inches, base64, **kwargs)
    148     from matplotlib.backend_bases import FigureCanvasBase
    149     FigureCanvasBase(fig)
--> 151 fig.canvas.print_figure(bytes_io, **kw)
    152 data = bytes_io.getvalue()
    153 if fmt == 'svg':

File ~\anaconda3\envs\conda310\lib\site-packages\matplotlib\backend_bases.py:2319, in FigureCanvasBase.print_figure(self, filename, dpi, facecolor, edgecolor, orientation, format, bbox_inches, pad_inches, bbox_extra_artists, backend, **kwargs)
   2315 try:
   2316     # _get_renderer may change the figure dpi (as vector formats
   2317     # force the figure dpi to 72), so we need to set it again here.
   2318     with cbook._setattr_cm(self.figure, dpi=dpi):
-> 2319         result = print_method(
   2320             filename,
   2321             facecolor=facecolor,
   2322             edgecolor=edgecolor,
   2323             orientation=orientation,
   2324             bbox_inches_restore=_bbox_inches_restore,
   2325             **kwargs)
   2326 finally:
   2327     if bbox_inches and restore_bbox:

File ~\anaconda3\envs\conda310\lib\site-packages\matplotlib\backend_bases.py:1648, in _check_savefig_extra_args.<locals>.wrapper(*args, **kwargs)
   1640     _api.warn_deprecated(
   1641         '3.3', name=name, removal='3.6',
   1642         message='%(name)s() got unexpected keyword argument "'
   1643                 + arg + '" which is no longer supported as of '
   1644                 '%(since)s and will become an error '
   1645                 '%(removal)s')
   1646     kwargs.pop(arg)
-> 1648 return func(*args, **kwargs)

File ~\anaconda3\envs\conda310\lib\site-packages\matplotlib\_api\deprecation.py:412, in delete_parameter.<locals>.wrapper(*inner_args, **inner_kwargs)
    402     deprecation_addendum = (
    403         f"If any parameter follows {name!r}, they should be passed as "
    404         f"keyword, not positionally.")
    405     warn_deprecated(
    406         since,
    407         name=repr(name),
   (...)
    410                  else deprecation_addendum,
    411         **kwargs)
--> 412 return func(*inner_args, **inner_kwargs)

File ~\anaconda3\envs\conda310\lib\site-packages\matplotlib\backends\backend_agg.py:540, in FigureCanvasAgg.print_png(self, filename_or_obj, metadata, pil_kwargs, *args)
    490 @_check_savefig_extra_args
    491 @_api.delete_parameter("3.5", "args")
    492 def print_png(self, filename_or_obj, *args,
    493               metadata=None, pil_kwargs=None):
    494     """
    495     Write the figure to a PNG file.
    496 
   (...)
    538         *metadata*, including the default 'Software' key.
    539     """
--> 540     FigureCanvasAgg.draw(self)
    541     mpl.image.imsave(
    542         filename_or_obj, self.buffer_rgba(), format="png", origin="upper",
    543         dpi=self.figure.dpi, metadata=metadata, pil_kwargs=pil_kwargs)

File ~\anaconda3\envs\conda310\lib\site-packages\matplotlib\backends\backend_agg.py:431, in FigureCanvasAgg.draw(self)
    429 def draw(self):
    430     # docstring inherited
--> 431     self.renderer = self.get_renderer(cleared=True)
    432     # Acquire a lock on the shared font cache.
    433     with RendererAgg.lock, \
    434          (self.toolbar._wait_cursor_for_draw_cm() if self.toolbar
    435           else nullcontext()):

File ~\anaconda3\envs\conda310\lib\site-packages\matplotlib\backends\backend_agg.py:447, in FigureCanvasAgg.get_renderer(self, cleared)
    444 reuse_renderer = (hasattr(self, "renderer")
    445                   and getattr(self, "_lastKey", None) == key)
    446 if not reuse_renderer:
--> 447     self.renderer = RendererAgg(w, h, self.figure.dpi)
    448     self._lastKey = key
    449 elif cleared:

File ~\anaconda3\envs\conda310\lib\site-packages\matplotlib\backends\backend_agg.py:93, in RendererAgg.__init__(self, width, height, dpi)
     91 self.width = width
     92 self.height = height
---> 93 self._renderer = _RendererAgg(int(width), int(height), dpi)
     94 self._filter_renderers = []
     96 self._update_methods()

ValueError: Image size of 1144x97463 pixels is too large. It must be less than 2^16 in each direction.

<Figure size 1152x288 with 1 Axes>
```

**Comment by ericpre** [comment]
It seems that the limit used are too large or something similar... can you provide a minimal reproducible example?

**Comment by ZGainsforth** [comment]
```
import numpy as np
import hyperspy.api as hs
S = hs.load("Globule bottom left.spx")
S.add_elements(['C', 'N', 'O', 'Na', 'Mg', 'Si', 'P', 'S', 'Cl', 'Ca', 'Fe', 'Ni', 'Cu', 'Ga', 'Pt'])
S.plot(True, norm='log')
```

However, it doesn't happen with every spectrum so it may be that the combination of spectrum and elements is necessary to trigger it.  I'm attaching a zip containing the failing bruker spx file and a jupyter notebook triggering the issue.

I'm running python 3.10

Thanks!

[Hyperspy minimal.zip](https://github.com/hyperspy/hyperspy/files/9418640/Hyperspy.minimal.zip)

**Comment by ericpre** [comment]
Thanks, this is a good example to reproduce.

**Comment by ericpre** [comment]
The issue is with the `inline` backend, if you use another matplotlib backend (`qt`, `ipympl`), then it works fine.

**Comment by ZGainsforth** [comment]
OK, I switched to nbagg and it is better.  Thank you!

**Comment by ericpre** [comment]
Fixed in #2995.
